src/poetry/repositories/link_sources/base.py

```diff
@@ -79,13 +83,18 @@ def link_package_data(self, link: Link) -> Package:
         with contextlib.suppress(ValueError):
             version = Version.parse(version)
 
-        return Package(name, version, source_url=link.url)
+        try:
+            return Package(name, version, source_url=link.url)
+        except Exception:
```

Comment:
```suggestion
        except InvalidVersion:
```

Reply:
Shouldn't we keep going no matter what exception we get? I think currently, only `InvalidVersion` can be raised, but the semantics of "if _anything_ goes wrong, ignore and continue" seem right to me.

WDYT?

Reply:
I think it's debatable.

On the one hand catching `Exception` seems more robust, but on the other hand it's an anti pattern that masks unexpected errors and makes debugging difficult (see [SO answer](https://stackoverflow.com/a/21950), [Real Python article](https://realpython.com/the-most-diabolical-python-antipattern/) or [Pylint Warning](https://pylint.pycqa.org/en/latest/messages/warning/broad-except.html)).

Since the only exception we are expecting is `InvalidVersion` and another exception may disguise something, I tend to being specific.

Reply:
By the way, we don't have to pass an invalid version to `Package` at all since we've already tried to parse some lines above. Instead of with `with contextlib.suppress(...)` we can do something like:

```
try:
    version = Version.parse(version)
except InvalidVersion:
    return None
return Package(name, version, source_url=link.url)
```